### DISCUSSION TITLE
Question: is there a way to test individual records if they comply with search params?

### DISCUSSION THREAD

**Discussion by mareczek** [activerecord-hackery/ransack#1629] [General]
Ransack is THE MOST valuable gem ever. It's absolutely great - thank you for creating and maintaining it!

It would be awesome if there was a method for testing individual records if they comply with the search params.

Having search params defined as:
```ruby
search_params: {
  name_cont: 'hello'
}
```

we use it as:
`Client.ransack(search_params).result`

which yields a beautiful sql query with results.

I would love to be able to perform a test on a individual record without touching the database as such:
`client.ransack_test(search_params) == true`

Is something like the above possible?

**Comment by slimdave** [comment]
Late to this issue, but while I don't think it's feasible to do this without touching the database, we do something similar by chaining on a call to `#exists?`.

```
irb(main):006:0> Book.ransack(title_cont: "phys").result.exists?(id: 186000)
=> true
irb(main):007:0> Book.ransack(title_cont: "phys").result.exists?(id: 186001)
=> false
```

This is generally going to be very efficient because the RDBMS is likely to rewrite the query execution plan.

**Comment by mareczek** [comment]
Hi @slimdave , thanks for that. I do indeed use `exists?` - very good utility that works perfectly with Ransack.
My use-case was a bit different. In my case, the entry might not yet be in the database or it is in the database but before the update.
In the example i shared, I have a instance variable `client` - which might be a new (not persisted) object or an object with not yet committed changes.

I'm absolutely certain that such a feature is not in the scope of Ransack. I just posted the question to see if others would also need such a functionality or if the Maintainers are doing something similar.

Thank you again :)
